### DISCUSSION TITLE
Local communication to robot via TCP 58867 required?

### DISCUSSION THREAD

**Discussion by Starbase12** [copystring/ioBroker.roborock#581] [Q&A] [answered]
Hello,

Because of lagging trust into proper security of IoT devices like vacuum cleaner my Rocky is connected through a separate **guest-network**. By design it's not possible for a device like my iobroker server to connect directly to the vacuum cleaner. But that seem to be required to get your adapter work correctly. Am I right?

I wish I could remote-start configured "programs" I setup earlier via Roborock app or do basic things like "pause", "resume" and "back-to-dock" via iobroker controls. The Roborock App itself can control the robot that way. Is there's a way your adapter can do the same?

**Comment by copystring** [comment, marked as answer]
Hi,

this used to be different. A few versions ago the connection was via mqtt only. However, roborock eventually introduced a rate limiter on their mqtt server resulting in timeouts until 24 hours passed. The only way to permanently make this work is via direct tcp connection to the robot.
I'm afraid it's out of my control and by the design of the mqtt server roborock is hosting.

**Reply by Starbase12** [reply]
Thx copystring and thx for your hard work!
